### PR TITLE
[CORE-5979] Discount Codes docs

### PR DESCRIPTION
**Pull request by ARtoriouSs**
Docs for DiscountCode and DiscountCodeUsage

Some demo screenshots:
* https://a.cl.ly/NQu4nA7Q
* https://a.cl.ly/mXubNWJo
* https://a.cl.ly/9Zuz69DB

### CODE REVIEW

File: content/reference/endpoints/booking/discount_codes.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+<div class="tab-content" markdown="1">
+<div class="tab-pane active" id="discount_code_read" markdown="1">
+Name                     | Type                             | Read/Write | Description
+-------------------------|----------------------------------|------------|---------------------------------------------
+id                       | String[UUID]                     | Read       | Discount Code's id.
+discount_code            | String                           | Read       | Discount Code itself.
+currency                 | String                           | Read       | Discount Code's currency.
+fixed_amount             | Float                            | Read       | Fixed amount of discount.
+percentage               | integer                          | Read       | Discount percentage.
+all_rentals              | Boolean                          | Read       | Flag, that indicates that this discount is applicable to all rentals with the same currency.
```
Comment (adamgrad) [inline content/reference/endpoints/booking/discount_codes.md:26]: with provided currency, 'same' could be a bit ambiguous in this context

File: content/reference/endpoints/booking/discount_codes.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+all_rentals              | Boolean                                                       | Read       | Flag, that indicates that this discount is applicable to all rentals with the same currency.
+number_of_times          | Integer                                                       | Read/Write | Number of times the Discount Code can be applied.
+minimum_nights_booked    | integer                                                       | Read/Write | Minimum number of nights that must be booked in order to apply Discount Code.
+minimum_spend            | Float                                                         | Read/Write | Minimum amount that must be spent in order to apply Discount Code.
+start_date               | [Date](/reference/enums#formats)                              | Read/Write | The start date from which the Discount Code will be applicable.
+end_date                 | [Date](/reference/enums#formats)                              | Read/Write | The expiration date of the Discount Code.
+rentals_eligibility      | [String](/reference/enums#discount-code-rentals-eligibility)  | Write      | What rentals this Discount Code can be applied to.
```
Comment (adamgrad) [inline content/reference/endpoints/booking/discount_codes.md:54]: good to mention here that this should remain empty if all_rentals is true. Also, if all rentals is true - array is ignored.

**Review by adamgrad — COMMENTED** [review]
@ARtoriouSs Overall looks great ! Just tiny suggestions, also were you able to render it locally? Would be good to share a demo screenshot for new sections in the PR description

File: content/reference/endpoints/booking/discount_codes.md (hunk 2 above)
Comment (ARtoriouSs) [reply content/reference/endpoints/booking/discount_codes.md:54]: Added a note that `rental_ids` should be empty when `rentals_eligibility` is `"all_rentals"`

**Review by adamgrad — APPROVED** [review]

**Comment by radekzawada** [comment]
@ARtoriouSs there is missing webhooks part, and what about extending bookings with discount code usage

**Review by radekzawada — APPROVED** [review]

**Pull request merged**